.github/workflows/deployBlocker.yml

```diff
@@ -31,7 +31,7 @@ jobs:
         run: |
           echo "DEPLOY_BLOCKER_URL=${{ github.event.pull_request.html_url }}" >> $GITHUB_ENV
           echo "DEPLOY_BLOCKER_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
-          echo "DEPLOY_BLOCKER_TITLE=${{ github.event.pull_request.title }}" >> $GITHUB_ENV
+          echo "DEPLOY_BLOCKER_TITLE=$(sed -e "s/'/'\\\\''/g; s/\`/\\\\\`/g; 1s/^/'/; \$s/\$/'/" <<< ${{ github.event.pull_request.title }})" >> $GITHUB_ENV
```

Comment:
Can you show whatever means you used to test this regex locally?

Reply:
So I did this to test it.

Created one text.txt file with some content E.g. 
```
[NO QA] fix: action failure on " " invalid 'String' \in `BASH`
```
Now in the terminal run
```
DEPLOY_BLOCKER_TITLE=$(sed -e "s/'/'\\\\''/g; s/\`/\\\\\`/g; 1s/^/'/; \$s/\$/'/" text.txt)
echo $DEPLOY_BLOCKER_TITLE
```
If you see the output and no error then it passed.

Reply:
We don't need to escape " else as we wrap our string within a single quote.

Reply:
So following your testing example, I got this:

```
'[NO QA] fix: action failure on " " invalid '\''String'\'' \in \`BASH\`'
```

But should it really be like this?

```
'[NO QA] fix: action failure on " " invalid \'String\' \in \`BASH\`'
```

Reply:
I just used one trick. Please read the first para of this answer https://stackoverflow.com/a/1315213. It is kind of preferred.

Reply:
😬 Jeez, bash is very counter-intuitive sometimes.

Reply:
But It's very powerful and I want to learn it.

Reply:
So today I learned about some bash magics that might be an easier / more generalized way to escape strings in bash. Basically you can:

1. Write a comment
1. Quote the last command in the bash history.

Try this in your terminal (I was careful to use bash not zsh to more closely reflect the GH environment):

```bash
> # this is a good example of special characters in bash - ' " ! ---------------
> !:q
```

Then check out the output! It seems like bash will automatically escape stuff this way and then we don't have to worry about using sed or a custom regex. The GHA code might look like this:

```
# ${{ github.event.pull_request.title }}
echo "DEPLOY_BLOCKER_TITLE=$(!:q) >> $GITHUB_ENV"
```

I would want to test this a bit, but seems better right?

Reply:
Ah, one hell of a nasty trick. But I am wondering how will we unescape it back to Normal before posting to slack. Bash Escaping is very strange.

Reply:
What do you think about that? IMO, escaping ourselves gives full control over it.

I tested it with 
```
It matches “badge.txt” because $the# wil* dcard repr$#@*&^%?><:":{} [];',./ \|esents a*ny (sequence)! ~ @of cha--racters+ or n= {o} c}har>a<c?te/rs. This command matches??  all? f`il`es called “source,” regardless of the file extension.
```
And all good. your thoughts?

Reply:
Okay, looking back at this, I think I agree that the comment trick is too hacky, so we can stick with using `sed`.

Reply:
This doesn't seem to work as expected, because the `<<< ${{ github.event.pull_request.title }}` part is expanded to the actual value of the variable when the script runs, but `<<<` only expects one word after it. If the PR title also contains a quote, then it breaks, see https://github.com/Expensify/App/commit/ff1d6f342436615fcccedb30638eb0c5ab79dcec/checks.

I opened https://github.com/Expensify/App/pull/4598 as a temporary fix, but it will also break if the PR title contains `'"`. Can't think of a better solution at the moment, open to any suggestions.